riff-cli/cmd/create.go

```diff
@@ -31,6 +31,16 @@ func Create(initCmd *cobra.Command, buildCmd *cobra.Command, applyCmd *cobra.Com
 	createChainCmd.Long = utils.CreateCmdLong()
 	createChainCmd.SetUsageTemplate(utils.CustomInvokerUsageTemplate)
 
+	// ignore all validation
+	//
+	// This command will fail since no invoker is specified. The init command
+	// is able to provide a more meaningful error message. We still want to
+	// use a chained command to establish persistant flags
+	createChainCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error { return nil }
```

Comment:
Not sure I understand this comment.

Who's "this" in the sentence "this command will fail"? Is it `createChainCmd`? If we have a command that will fail, should we have a command in the first place?


Persistence of flags has to do with "inheritance" of commands. Chaining is more akin to "composition" of commands.
I'm more and more thinking that our implementation of chaining brings more complexity than actually needed.

Reply:
`createChainCmd` is the command that will fail, since it doesn't have an invoker specified. The purpose of the command is twofold:
- provide a good error message when a user executes the command, either by omitting an invoker or specifying an invoker that is not installed
- generate command docs that aggregate behavior across invokers

I agree it's a bit funny to define a command that we know will never do real work. If there's another way to accomplish these goals, I'm in.